activerecord/lib/active_record/relation/from_clause.rb

```diff
@@ -25,7 +25,7 @@ def empty?
       end
 
       def self.empty
-        new(nil, nil)
+        @empty ||= new(nil, nil).freeze
```

Comment:
`freeze` is added here for the purpose of validating that indeed this object is never mutated (and thus is safe to memoize), and the same is true at the other location. I suppose I can remove the call to `freeze`, but I don't see the upside. It may be obvious to you that these objects happen to be immutable because you know this code, but the presence of `freeze` here makes that fact obvious to uninitiated readers as well.